dipy/reconst/mapmri.py

```diff
@@ -0,0 +1,666 @@
+                 eigenvalue_threshold=1e-04,
+                 bmax_threshold=2000):
+        r""" Analytical and continuous modeling of the diffusion signal with
+        respect to the MAPMRI basis [1]_.
+
+        The main idea is to model the diffusion signal as a linear combination of
+        the continuous functions presented in [2]_ but extending it in three
+        dimension.
+
+        The main difference with the SHORE proposed in [3]_ is that MAPMRI 3D
+        extension is provided using a set of three basis function for the radial
+        part, one for the signal along x, one for y and one for z, while [3]_ 
+        uses one basis function to model the radial part and real Spherical
+        Harmonics to model the angular part.
+        From the MAPMRI coefficients is possible to use the analytical formulae 
+        to estimate the ODF.
+
+
+        Parameters
+        ----------
+        gtab : GradientTable,
+            gradient directions and bvalues container class
+        radial_order : unsigned int,
+            an even integer that represent the order of the basis
+        lambd : float,
+            radial regularisation constant
+        eap_cons : bool,
+            Constrain the propagator to be positive.
+        anisotropic_scaling : bool,
+            If false, force the basis function to be identical in the three
+            dimensions (SHORE like).
+        eigenvalue_threshold : float,
+            set the minimum of the tensor eigenvalues in order to avoid 
+            stability problem
+        bmax_threshold : float,
+            set the maximum b-value for the tensor estimation
+
+        References
+        ----------
+        .. [1] Ozarslan E. et. al, "Mean apparent propagator (MAP) MRI: A novel
+               diffusion imaging method for mapping tissue microstructure",
+               NeuroImage, 2013.
+
+        .. [2] Ozarslan E. et. al, "Simple harmonic oscillator based reconstruction
+               and estimation for one-dimensional q-space magnetic resonance
+               1D-SHORE)", eapoc Intl Soc Mag Reson Med, vol. 16, p. 35., 2008.
+
+        .. [3] Ozarslan E. et. al, "Simple harmonic oscillator based reconstruction
+               and estimation for three-dimensional q-space mri", ISMRM 2009.
+
+        Examples
+        --------
+        In this example, where the data, gradient table and sphere tessellation
+        used for reconstruction are provided, we model the diffusion signal
+        with respect to the MAPMRI model and compute the real and analytical
+        ODF.
+
+        >>> from dipy.core.gradients import gradient_table
+        >>> from dipy.data import dsi_voxels, get_sphere
+        >>> data, gtab = dsi_voxels()
+        >>> sphere = get_sphere('symmetric724')
+        >>> from dipy.sims.voxel import SticksAndBall
+        >>> data, golden_directions = SticksAndBall(gtab, d=0.0015, S0=1, angles=[(0, 0), (90, 0)], fractions=[50, 50], snr=None)
```

Comment:
I think it should be possible to brake this line so that it doesn't take so much space.